apps/safe-claiming-app/src/Widget.tsx

```diff
@@ -0,0 +1,196 @@
+  )
+
+  const [ecosystemVestingStatus] = useFetchVestingStatus(
+    ecosystemVesting?.vestingId,
+    ecosystemVesting?.contract
+  )
+
+  const [investorVestingStatus] = useFetchVestingStatus(
+    investorVesting?.vestingId,
+    investorVesting?.contract
+  )
+
+  const totalClaimed = BigNumber.from(userVestingStatus?.amountClaimed || 0)
+    .add(ecosystemVestingStatus?.amountClaimed || 0)
+    .add(investorVestingStatus?.amountClaimed || 0)
+
+  const votingPower = useMemo(() => {
+    if (totalAllocation.gt(0) && totalClaimed.gt(0) && balance.gt(0)) {
+      return totalAllocation.sub(totalClaimed).add(balance)
+    }
+    return BigNumber.from(0)
+  }, [balance, totalAllocation, totalClaimed])
+
+  const ctaWidget = (
+    <SpaceContent>
+      <div>
+        <Title>Become the part of Safe future!</Title>
+        <br />
+        <Subtitle>
+          You will be able to buy $SAFE once the token transferability is
+          enabled.
+        </Subtitle>
+      </div>
+      <StyledButton disabled>Buy Tokens</StyledButton>
+    </SpaceContent>
+  )
+
+  const votingPowerWidget = (
+    <SpaceContent>
+      <div>
+        <Title>Your voting power</Title>
+        <Box
+          display="flex"
+          alignItems="center"
+          justifyContent="center"
+          gap={1}
+          mb={1}
+        >
+          <SafeIcon />
+          <Typography variant="h5">
+            {formatAmount(Number(ethers.utils.formatEther(votingPower)), 2)}{" "}
+          </Typography>
+        </Box>
+      </div>
+      {totalClaimed.gt(0) ? (
+        <>
+          <Subtitle>
+            You've already claimed{" "}
+            {formatAmount(Number(ethers.utils.formatEther(totalClaimed)), 2)}{" "}
+            SAFE
+          </Subtitle>
+          {currentDelegate && (
+            <Box width={1}>
+              <Typography variant="caption" marginBottom={1}>
+                Delegated to
+              </Typography>
+              <SelectedDelegate delegate={currentDelegate} onClick={() => {}} />
+            </Box>
+          )}
+        </>
+      ) : (
+        <>
+          <Subtitle>
+            Claim your tokens to start participating in voting
+          </Subtitle>
+          <Link
+            href={`${window.location.ancestorOrigins[0]}/apps?safe=${
+              safe.chainId === 1 ? "eth" : "gor"
+            }:${safe.safeAddress}&appUrl=${CLAIMING_APP_URL}`}
+            rel="noopener noreferrer"
+            target="_blank"
+            display="inline-flex"
+            alignItems={"center"}
+            underline="none"
+          >
+            <StyledButton>Claim And Delegate</StyledButton>
+          </Link>
+        </>
+      )}
+    </SpaceContent>
+  )
+
+  return (
+    <Box
+      display="flex"
+      flexDirection="column"
+      alignItems="center"
+      height="300px"
+      p={3}
+      sx={{ backgroundColor: "background.paper" }}
+    >
+      {votingPower.eq(0) ? ctaWidget : votingPowerWidget}
```

Comment:
While loading data, the app flickers and briefly shows "Become part of the Safe future".

I would suggest that we use the loading states of the used hooks and probably introduce a loading state into the useTokenBalance hook to display Skeletons while loading.